model/task/task.go

```diff
@@ -3177,6 +3189,15 @@ func AddParentDisplayTasks(tasks []Task) ([]Task, error) {
 func (t *Task) UpdateDependsOn(status string, newDependencyIDs []string) error {
 	newDependencies := make([]Dependency, 0, len(newDependencyIDs))
 	for _, depID := range newDependencyIDs {
+		if depID == t.Id {
+			grip.Error(message.Fields{
+				"message": "task is attempting to add a dependency on itself, skipping this dependency",
```

Comment:
Since we can't stop on error, logging the error is the whole point. Our only alternative is to ignore the error completely, which is bad for traceability, since trying to add a self dependency is a bug (i.e. if this log is ever hit, someone tried to update the dependencies without checking the validity of the dependency graph).